agents/agents-features/agents-features-opentelemetry/src/jvmMain/kotlin/ai/koog/agents/features/opentelemetry/feature/OpenTelemetry.kt

```diff
@@ -226,20 +236,10 @@ public class OpenTelemetry {
                 spanProcessor.startSpan(inferenceSpan)
 
                 // Add events to the InferenceSpan after the span is created
-                val lastMessage = eventContext.prompt.messages.lastOrNull()
-
-                val events: List<GenAIAgentEvent> = lastMessage?.let { message ->
-                    buildList {
-                        when (message) {
-                            is Message.User -> add(UserMessageEvent(provider, message, verbose = config.isVerbose))
-                            is Message.System -> add(SystemMessageEvent(provider, message, verbose = config.isVerbose))
-                            is Message.Assistant -> add(AssistantMessageEvent(provider, message, verbose = config.isVerbose))
-                            else -> {}
-                        }
-                    }
-                } ?: emptyList()
-
-                inferenceSpan.addEvents(events)
+                eventContext.prompt.messages.forEachIndexed { index, message ->
+                    inferenceSpan.span.setAttribute("gen_ai.prompt.$index.role", message.role.name.lowercase())
```

Comment:
Could you please use dedicated classes for each event type instead of adding attributes in a "raw way". It helps to keep code more consistent.